src/test/java/com/list/todo/services/TodoListServiceTest.java

```diff
@@ -219,25 +289,80 @@ public void deleteSharedTodoList_Successful() {
         verify(shareService).isSharedTodoList(TODO_LIST_ID);
         verify(shareService).deleteShareBySharedTodoListId(TODO_LIST_ID);
         verify(todoListRepository, never()).deleteById(TODO_LIST_ID);
+        verify(notificationService, never()).notifyAboutDeletingTodoList(user, todoList);
     }
 
     @Test
-    public void shareTodoList_Successful() {
+    public void shareTodoList_ShareExistentTodoListToExistentUser_ApiResponseWithSuccess() {
         //arrange
+        ApiResponse apiResponse = new ApiResponse(true, "You shared your todoList to " + SECOND_USER_USERNAME + "!");
+        User user = new User();
         User targetUser = new User();
+        targetUser.setId(SECOND_USER_ID);
         TodoList todoList = new TodoList();
         todoList.setTodoListName(TODO_LIST_NAME);
 
-        when(todoListRepository.findById(TODO_LIST_ID_2)).thenReturn(Optional.of(todoList));
+        when(todoListRepository.findById(TODO_LIST_ID)).thenReturn(Optional.of(todoList));
         when(userService.getUserByUsername(SECOND_USER_USERNAME)).thenReturn(Optional.of(targetUser));
+        when(userService.getUserById(USER_ID)).thenReturn(Optional.of(user));
 
         //act
-        todoListService.shareTodoList(SECOND_USER_USERNAME, TODO_LIST_ID, USER_ID);
+        ApiResponse result = todoListService.shareTodoList(SECOND_USER_USERNAME, TODO_LIST_ID, USER_ID);
         Share share = new Share(targetUser.getId(), todoList);
 
         //assert
-        verify(todoListRepository).findById(SECOND_USER_ID);
-        verify(shareService).addShare(share);
+        Assert.assertEquals(apiResponse, result);
+        verify(todoListRepository, times(1)).findById(TODO_LIST_ID);
+        verify(userService, times(1)).getUserByUsername(SECOND_USER_USERNAME);
+        verify(shareService, times(1)).addShare(share);
+        verify(userService, times(1)).getUserById(USER_ID);
+        verify(notificationService, times(1)).notifyAboutSharingTodoList(user, targetUser, todoList);
+    }
+
+    @Test
+    public void shareTodoList_ShareNonExistentTodoListToExistentUser_ApiResponseWithFailure() {
+        //arrange
+        ApiResponse apiResponse = new ApiResponse(false, "Something went wrong!");
```

Comment:
As far as I understand you don't use ApiResponse in production code as it was intended.
Add //TODO comment onto ApiResponse class to handle this issue when refactoring